feature(examples): add awardco connector (#266)
fivetran/fivetran_connector_sdk: examples/source_examples/awardco/README.md

> # AwardCo Users Connector

Reviewer: ```suggestion
# AwardCo Users Connector Example
```

AwardCo is an employee recognition platform. The AwardCo API typically exposes RESTful JSON endpoints to manage resources such as users and recognition-related entities. Authentication is performed via an API key supplied in request headers. Common behaviors include paginated responses, timestamp fields for change tracking, and conventional HTTP status codes. Refer to official AwardCo documentation for authoritative details and production configurations.

## Connector overview

> This example connector uses the Fivetran Connector SDK to sync AwardCo user data into your destination. It performs incremental syncs based on a timestamp cursor, upserts user rows into a single `user` table, and emits checkpoints for reliable resumption. A local mock mode is provided for offline development and testing.

Reviewer 2: ```suggestion
This example connector uses the Fivetran Connector SDK to sync AwardCo user data into your destination. It performs incremental syncs based on a timestamp cursor, upserts user rows into a single `USER` table, and emits checkpoints for reliable resumption. There is also a local mock mode for offline development and testing.
```

Related code
- Configuration validation — `awardco-users-connector/connector.py:13`
- Schema definition — `awardco-users-connector/connector.py:26`
- Sync/update logic — `awardco-users-connector/connector.py:47`

## Requirements

> - Python 3.9–3.12

Reviewer 2: ```suggestion
- [Supported Python versions](https://github.com/fivetran/fivetran_connector_sdk/blob/main/README.md#requirements)
```

Validation rules
- `api_key` (required): AwardCo API key.
- `base_url` (required): AwardCo API base URL. Do not include a trailing `/api` because the connector appends the endpoint path (for example, `/api/users`).
- `rate_limit` (optional): User-provided hint; not enforced by code.
- `use_mock` (optional): When `true`, reads `files/mock_users.json` instead of calling the API.

Implementation reference
- Validation — `awardco-users-connector/connector.py:13`
- Config usage — `awardco-users-connector/connector.py:58`

## Requirements file
This example does not require external libraries at runtime for the connector itself. The provided `requirements.txt` is intentionally minimal.

Example (optional dev tooling):

```
# requirements.txt
# Add optional tools only if needed for local development
# Faker can be used to generate mock data, but is not required by the connector
Faker
```

Note: The `fivetran_connector_sdk` and `requests` packages are pre-installed in the Fivetran environment. Do not declare them in `requirements.txt` to avoid conflicts.

## Authentication
This connector uses an API key. The request includes the key in the `apiKey` header when calling the AwardCo API.

> Steps

Reviewer: To obtain the API key, do the following:

- Obtain an API key from your AwardCo administrator or developer portal.

> - Set `api_key` in `configuration.json`.

Reviewer: ```suggestion
2. Set `api_key` in `configuration.json`.
```

> - Set `base_url` (for example, `https://api.awardco.com`).

Reviewer: ```suggestion
3. Set `base_url` (for example, `https://api.awardco.com`).
```
